src/particle-analysis/EventCharacteristics.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+from ParticleClass import Particle
+from Lattice3D import Lattice3D
+
+class EventCharacteristics:
+    """
+    This class computes event-by-event characteristics, e.g., eccentricities
+    or certain densities.
+
+    Parameters
+    ----------
+    particle_data: list, numpy.ndarray
+        List or array containing particle objects for one event.
+
+    Attributes
+    ----------
+    particle_data_: list, numpy.ndarray
+        List or array containing particle objects for one event.
+
+    Methods
+    -------
+    eccentricity_from_particles:
+        Computes the spatial eccentricity from particles.
+    """
+    def __init__(self,particle_data):
+        # check that the input is a list/numpy.ndarray containing Particle objects
+        if not isinstance(particle_data, (list, np.ndarray)):
+            raise TypeError('The input is not a list nor a numpy.ndarray.')
+        for particle in particle_data:
+            if not isinstance(particle, Particle):
+                raise TypeError('At least one element in the input is not a ' +\
+                                'Particle type.')
+        self.particle_data_ = particle_data
+
+    
+    def eccentricity_from_particles(self,harmonic_n):
+        real_eps=0
+        imag_eps=0
+        norm=0
+        for particle in particle_data:
+            x = particle.x
+            y = particle.y
+            rn = x**harmonic_n+y**harmonic_n
+            t = np.arctan2(y,x)
+            real_eps += rn*np.cos(harmonic_n*t)*particle.E
```

Comment:
For the particles it would be possible to have the option to weight the eccentricities by the particles (`particles.E`  $\rightarrow$ `1`) and not by the energy. This could be an option in the function call.

Reply:
I think when we are at it, we should also allow charge and baryon number weighting. It is not very common but there are usecases.